From 6212a5ec03bdcfea197ca205c3ca06c2d275c41e Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Mon, 26 Aug 2013 17:08:52 +0200
Subject: [PATCH 1/4] Revert "chmod/chown envs dir when preparing this"

This reverts commit 137e683760707c690df496516432d72d8f7a81d3.

In Fedora our basedir is /var/cache/icecream, it's shipped in the rpm
and its owned by root:icecc. We cannot and do not want to chown/chmod it.
---
 daemon/environment.cpp | 7 +------
 daemon/environment.h   | 2 +-
 daemon/main.cpp        | 2 +-
 3 files changed, 3 insertions(+), 8 deletions(-)

diff --git a/daemon/environment.cpp b/daemon/environment.cpp
index 4cfb504e39..fac3334754 100644
--- a/daemon/environment.cpp
+++ b/daemon/environment.cpp
@@ -162,7 +162,7 @@ static bool cleanup_directory(const string &directory)
     return true;
 }
 
-bool cleanup_cache(const string &basedir, uid_t user_uid, gid_t user_gid)
+bool cleanup_cache(const string &basedir)
 {
     flush_debug();
 
@@ -181,11 +181,6 @@ bool cleanup_cache(const string &basedir, uid_t user_uid, gid_t user_gid)
         return false;
     }
 
-    if (chown(basedir.c_str(), user_uid, user_gid) || chmod(basedir.c_str(), 0775)) {
-        log_perror("chown/chmod in cleanup_cache() failed") << "\t" << basedir << endl;;
-        return false;
-    }
-
     return true;
 }
 
diff --git a/daemon/environment.h b/daemon/environment.h
index dd61e7104b..9fbd331c5b 100644
--- a/daemon/environment.h
+++ b/daemon/environment.h
@@ -29,7 +29,7 @@
 #include <unistd.h>
 
 class MsgChannel;
-extern bool cleanup_cache(const std::string &basedir, uid_t user_uid, gid_t user_gid);
+extern bool cleanup_cache(const std::string &basedir);
 extern int start_create_env(const std::string &basedir,
                             uid_t user_uid, gid_t user_gid,
                             const std::string &compiler, const std::list<std::string> &extrafiles,
diff --git a/daemon/main.cpp b/daemon/main.cpp
index b0dbd95ee4..9fe151a275 100644
--- a/daemon/main.cpp
+++ b/daemon/main.cpp
@@ -2535,7 +2535,7 @@ int main(int argc, char **argv)
     pidFile << dcc_master_pid << endl;
     pidFile.close();
 
-    if (!cleanup_cache(d.envbasedir, d.user_uid, d.user_gid)) {
+    if (!cleanup_cache(d.envbasedir)) {
         return 1;
     }
 
-- 
2.21.0

